### PR TITLE
Yukon Employment Amount

### LINKED ISSUE #285: Canada Employment Amount
Reference: https://www.canada.ca/en/revenue-agency/services/tax/individuals/topics/about-your-tax-return/tax-return/completing-a-tax-return/deductions-credits-expenses/line-31260-canada-employment-amount.html

### PR DESCRIPTION
**Pull request by sqw0418**
Fixes #285

### CODE REVIEW

File: policyengine_canada/variables/gov/provinces/yt/benefits/yukon_employment_benefit/yt_employment_benefit.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
+from policyengine_canada.model_api import *
+
+
+class yt_employment_benefit(Variable):
```
Comment (PavelMakarchuk) [inline policyengine_canada/variables/gov/provinces/yt/benefits/yukon_employment_benefit/yt_employment_benefit.py:4]:
```suggestion
class yt_employment_amount(Variable):
```

Comment (PavelMakarchuk) [reply policyengine_canada/variables/gov/provinces/yt/benefits/yukon_employment_benefit/yt_employment_benefit.py:4]: need to change to credits
+ other folders / files

File: policyengine_canada/parameters/gov/provinces/yt/tax/income/credits/yt_employment_amount/yt_employment_amount.yaml
Comment (PavelMakarchuk) [inline policyengine_canada/parameters/gov/provinces/yt/tax/income/credits/yt_employment_amount/yt_employment_amount.yaml:1]: can you add the legal reference

Comment (PavelMakarchuk) [reply policyengine_canada/parameters/gov/provinces/yt/tax/income/credits/yt_employment_amount/yt_employment_amount.yaml:1]: add the appropriate page from [this](https://laws.yukon.ca/cms/images/LEGISLATION/PRINCIPAL/2002/2002-0118/2002-0118.pdf) document

File: policyengine_canada/parameters/gov/provinces/yt/tax/income/credits/yt_employment_amount/yt_employment_amount.yaml [hunk 2]
```diff
@@ -0,0 +1,9 @@
+description: Yukon provides this employment amount. 
+values:
+  2022-01-01: 1_287
+metadata:
+  unit: currency-CAD
+  label: Yukon Employment Amount
+  reference:
+  - title: Canada Employment Amount Benefit for Yukon
```
Comment (PavelMakarchuk) [inline policyengine_canada/parameters/gov/provinces/yt/tax/income/credits/yt_employment_amount/yt_employment_amount.yaml:8]: need to change to actual title of the document + add the yukon income tax act reference

Comment (PavelMakarchuk) [reply policyengine_canada/parameters/gov/provinces/yt/tax/income/credits/yt_employment_amount/yt_employment_amount.yaml:8]: in both references we need to specify the paragraph and page where the value is present

File: policyengine_canada/parameters/gov/provinces/yt/tax/income/credits/yt_employment_amount/yt_employment_amount.yaml [hunk 3]
```diff
@@ -0,0 +1,9 @@
+description: Yukon provides this employment amount. 
+values:
+  2022-01-01: 1_287
+metadata:
+  unit: currency-CAD
+  label: Yukon Employment Amount
```
Comment (PavelMakarchuk) [inline policyengine_canada/parameters/gov/provinces/yt/tax/income/credits/yt_employment_amount/yt_employment_amount.yaml:6]:
```suggestion
  label: Yukon employment benefit amount
```

File: policyengine_canada/parameters/gov/provinces/yt/tax/income/credits/yt_employment_amount/yt_employment_amount.yaml [hunk 4]
```diff
@@ -0,0 +1,9 @@
+description: Yukon provides this employment amount.
```
Comment (PavelMakarchuk) [inline policyengine_canada/parameters/gov/provinces/yt/tax/income/credits/yt_employment_amount/yt_employment_amount.yaml:1]:
```suggestion
description: Yukon provides this employment benefit amount. 
```

File: policyengine_canada/tests/gov/provinces/yt/tax/income/credits/yt_employment_amount/yt_employment_amount.yaml [hunk 5]
```diff
@@ -0,0 +1,24 @@
+- name: tax return with maximum amount
+  period: 2022
+  input:
+    province_code: YT
+    employment_income: 30000
+  output:
+    yt_employment_amount: 1_287
+
+- name: tax return for no income
```
Comment (PavelMakarchuk) [inline policyengine_canada/tests/gov/provinces/yt/tax/income/credits/yt_employment_amount/yt_employment_amount.yaml:9]:
```suggestion
- name: tax return with no income
```

File: policyengine_canada/tests/gov/provinces/yt/tax/income/credits/yt_employment_amount/yt_employment_amount.yaml [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+  period: 2022
+  input:
+    province_code: YT
+    employment_income: 0
+  output:
+    yt_employment_amount: 0
+
+- name: tax return for income under maximum amount
```
Comment (PavelMakarchuk) [inline policyengine_canada/tests/gov/provinces/yt/tax/income/credits/yt_employment_amount/yt_employment_amount.yaml:17]:
```suggestion
- name: tax return with income under maximum amount
```

File: policyengine_canada/variables/gov/provinces/yt/tax/yukon_employment_benefit/yt_employment_benefit.py [hunk 7]
```diff
@@ -0,0 +1,20 @@
+from policyengine_canada.model_api import *
+
+
+class yt_employment_amount(Variable):
+    value_type = float
+    entity = Person
+    label = "Yukon Employment Amount"
```
Comment (PavelMakarchuk) [inline policyengine_canada/variables/gov/provinces/yt/tax/yukon_employment_benefit/yt_employment_benefit.py:7]:
```suggestion
    label = "Yukon Employment Benefit Amount"
```

File: policyengine_canada/variables/gov/provinces/yt/tax/yukon_employment_benefit/yt_employment_benefit.py [hunk 8]
```diff
@@ -0,0 +1,20 @@
...
+    unit = CAD
+    definition_period = YEAR
+    defined_for = ProvinceCode.YT
+    reference = "https://www.canada.ca/en/revenue-agency/services/forms-publications/tax-packages-years/general-income-tax-benefit-package/yukon/5011-pc/information-residents-yukon.html#P4_58310"
+
+    def formula(person, period, parameters):
+        maximun_return_amount = parameters(
```
Comment (PavelMakarchuk) [inline policyengine_canada/variables/gov/provinces/yt/tax/yukon_employment_benefit/yt_employment_benefit.py:14]:
```suggestion
        max_amount = parameters(
```

File: policyengine_canada/variables/gov/provinces/yt/tax/income/benefits/yukon_employment_amount/yt_employment_amount.py [hunk 9]
```diff
@@ -0,0 +1,20 @@
...
+    definition_period = YEAR
+    defined_for = ProvinceCode.YT
+    reference = "https://www.canada.ca/en/revenue-agency/services/forms-publications/tax-packages-years/general-income-tax-benefit-package/yukon/5011-pc/information-residents-yukon.html#P4_58310"
+
+    def formula(person, period, parameters):
+        max_amount = parameters(
+            period
+        ).gov.provinces.yt.benefits.yt_employment_amount.yt_employment_amount
+
+        employment_income = person("employment_income", period)
```
Comment (PavelMakarchuk) [inline policyengine_canada/variables/gov/provinces/yt/tax/income/benefits/yukon_employment_amount/yt_employment_amount.py:18]:
```suggestion
        canada_employment_amount = person("cananda_employment_amount", period)
```

Comment (PavelMakarchuk) [reply policyengine_canada/variables/gov/provinces/yt/tax/income/benefits/yukon_employment_amount/yt_employment_amount.py:18]: need changes in test

File: policyengine_canada/parameters/gov/provinces/yt/benefits/yt_employment_amount/yt_employment_amount.yaml [hunk 10]
```diff
@@ -0,0 +1,11 @@
+description: Yukon provides this employment benefit amount. 
+values:
+  2022-01-01: 1_287
+metadata:
+  unit: currency-CAD
+  label: Yukon employment benefit amount
+  reference:
+  - title: Canada Employment Amount Benefit for Yukon -Page 33
+    href: https://laws.yukon.ca/cms/images/LEGISLATION/PRINCIPAL/2002/2002-0118/2002-0118.pdf
```
Comment (PavelMakarchuk) [inline policyengine_canada/parameters/gov/provinces/yt/benefits/yt_employment_amount/yt_employment_amount.yaml:9]:
```suggestion
    href: https://laws.yukon.ca/cms/images/LEGISLATION/PRINCIPAL/2002/2002-0118/2002-0118.pdf#page=33
```

File: policyengine_canada/parameters/gov/provinces/yt/benefits/yt_employment_amount/yt_employment_amount.yaml [hunk 11]
```diff
@@ -0,0 +1,11 @@
+description: Yukon provides this employment benefit amount. 
+values:
+  2022-01-01: 1_287
+metadata:
+  unit: currency-CAD
+  label: Yukon employment benefit amount
+  reference:
+  - title: Canada Employment Amount Benefit for Yukon -Page 33
```
Comment (PavelMakarchuk) [inline policyengine_canada/parameters/gov/provinces/yt/benefits/yt_employment_amount/yt_employment_amount.yaml:8]:
```suggestion
  - title: Yukon income tax act - "add paragraph"
```

File: policyengine_canada/parameters/gov/provinces/yt/benefits/yt_employment_amount/yt_employment_amount.yaml (hunk 10 above)
Comment (PavelMakarchuk) [reply policyengine_canada/parameters/gov/provinces/yt/benefits/yt_employment_amount/yt_employment_amount.yaml:9]: not sure if this is the right page

File: policyengine_canada/parameters/gov/provinces/yt/benefits/yt_employment_amount/yt_employment_amount.yaml [hunk 12]
```diff
@@ -0,0 +1,11 @@
...
+    href: https://laws.yukon.ca/cms/images/LEGISLATION/PRINCIPAL/2002/2002-0118/2002-0118.pdf
+  - title: Canada Employment Amount Benefit for Yukon
```
Comment (PavelMakarchuk) [inline policyengine_canada/parameters/gov/provinces/yt/benefits/yt_employment_amount/yt_employment_amount.yaml:10]:
```suggestion
  - title: Government of Canada - Yukon Information Guide
```

File: policyengine_canada/tests/gov/provinces/yt/tax/income/benefits/yt_employment_amount/yt_employment_amount.yaml
Comment (PavelMakarchuk) [inline policyengine_canada/tests/gov/provinces/yt/tax/income/benefits/yt_employment_amount/yt_employment_amount.yaml:1]: move file to credits folder

File: policyengine_canada/parameters/gov/provinces/yt/tax/income/credits/employment_amount/amount.yaml [hunk 13]
```diff
@@ -0,0 +1,12 @@
+description: Yukon provides this employment amount. 
+values:
+  2022-01-01: 1_287
+metadata:
+  unit: currency-CAD
+  label: Yukon employment amount
+  reference:
+  - title: Yukon Tax Form YT428 - Line 58310
+    href: https://www.canada.ca/content/dam/cra-arc/formspubs/pbg/5011-c/5011-c-22e.pdf#page=1
+    #Credit is not mentioned in the Yukon income tax act
```
Comment (PavelMakarchuk) [inline policyengine_canada/parameters/gov/provinces/yt/tax/income/credits/employment_amount/amount.yaml:10]:
```suggestion
    # Credit is not mentioned in the Yukon income tax act
```

**Review by PavelMakarchuk — APPROVED** [review]

**Review by MaxGhenis — APPROVED** [review]

**Pull request merged**